### PR TITLE
Restore `clean-header-search-field` for some users

### PR DESCRIPTION
**Pull request by fregante**
- Fixes https://github.com/refined-github/refined-github/issues/6310

## Test URLs

https://github.com/fregante/GhostText/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc

## Screenshot

Untested, depends on the user

cc @kaste

### CODE REVIEW

File: source/features/clean-header-search-field.tsx [hunk 1]
```diff
@@ -5,7 +5,14 @@ import select from 'select-dom';
 import features from '../feature-manager';
 
 async function init(): Promise<void> {
-	const headerSearchButton = await elementReady('button.header-search-button');
+	const headerSearchButton = await elementReady('button.header-search-button, input.header-search-input');
+	if (headerSearchButton instanceof HTMLInputElement) {
+		// Previous version #6310
+		// TODO: Remove in mid 2023
```
Comment (yakov116) [inline source/features/clean-header-search-field.tsx:11]: Maybe put a real date so it shows up as a warning

Comment (fregante) [reply source/features/clean-header-search-field.tsx:11]: I know we've been doing that, but what does that change in practice? We don't want CI to start failing and force us to immediately take action 🤔

**Comment by auscompgeek** [comment]
> Untested, depends on the user

It should be toggleable in the feature preview settings:

image

File: source/features/clean-header-search-field.tsx (hunk 1 above)
Comment (yakov116) [reply source/features/clean-header-search-field.tsx:11]: If I am not mistaken we set it to warn

Comment (fregante) [reply source/features/clean-header-search-field.tsx:11]: We did, but due to: 

- https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1540

I think `expiring-todo-comments` only makes sense:

- if it never breaks (on a date) and forces anyone to act immediately
- if someone is actively looking for and fixing TODO comments 
- if there are no other TODOs that are _also_ waiting for a fix anyway, for months

Essentially I think we can do away with them. Moving this discussion upstream in https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1541

**Pull request merged**